dateparser/parser.py

```diff
@@ -564,9 +564,19 @@ def _correct_for_time_frame(self, dateobj, tz):
             try:
                 tz = tz or get_timezone_from_tz_string(self.settings.TIMEZONE)
                 tz_offset = tz.utcoffset(dateobj)
-            except pytz.UnknownTimeZoneError:
+            except (pytz.UnknownTimeZoneError, pytz.NonExistentTimeError):
                 tz_offset = timedelta(hours=0)
 
+            dateobj_time = None
+            if tz:
+                try:
+                    dateobj_time = (dateobj - tz.utcoffset(dateobj)).time()
+                except pytz.InvalidTimeError:
+                    pass
+
+            if not dateobj_time:
+                dateobj_time = dateobj.time()
```

Comment:
This code looks like a no-op at the moment, it defines a `dateobj_time` variable that is never used.